src/commands/beer_style.rs

```diff
@@ -0,0 +1,100 @@
+extern crate clap;
+use clap::{App, Arg, ArgMatches, SubCommand};
+use crate::AppSubCommand;
+pub use crate::calculators::abv::Abv;
+pub use crate::utils::beer_styles::{BEER_STYLES, BeerStyle};
+
+pub struct BeerStyleFinder;
+
+impl AppSubCommand for BeerStyleFinder {
+    fn add_subcommand<'a, 'b>(&self, app: App<'a, 'b>) -> App<'a, 'b>{
+        let ret = app.subcommand(SubCommand::with_name("beer_style")
+            .version("0.1")
+            .author("Heikki Hellgren (heiccih@gmail.com)")
+            .about("Finds matches of beer style based on parameters")
+            .arg(Arg::with_name("og")
+                 .short("o")
+                 .long("og")
+                 .value_name("OG")
+                 .help("Original gravity")
+                 .takes_value(true))
+            .arg(Arg::with_name("fg")
+                 .short("f")
+                 .long("fg")
+                 .value_name("FG")
+                 .help("Final gravity")
+                 .takes_value(true))
+            .arg(Arg::with_name("abv")
+                 .short("a")
+                 .long("abv")
+                 .value_name("ABV")
+                 .help("Alcohol By Volume")
+                 .takes_value(true))
+            .arg(Arg::with_name("ibu")
+                 .short("i")
+                 .long("ibu")
+                 .value_name("IBU")
+                 .help("International Bittering Units")
+                 .takes_value(true))
+            .arg(Arg::with_name("color")
+                 .short("c")
+                 .long("color")
+                 .value_name("COLOR")
+                 .help("Standard Rererence Model Color")
+                 .takes_value(true)));
+        ret
+    }
+
+    fn do_matches<'a>(&self, matches: &ArgMatches<'a>) {
+        if let Some(ref matches) = matches.subcommand_matches("beer_style") {
+            let fg = matches.value_of("fg");
```

Comment:
I think we can use `if let` for this too, something like:
```
if let Some(fg) = matches.value_of("fg") {
...
}
```
instead of this and the check below.

Reply:
Fixed those!